### PR TITLE
Identity manager models

### PR DESCRIPTION
**Pull request by GhTsovinar**

### CODE REVIEW

File: src/IdentityManager/Travely.IdentityManager.Repository/UnitOfWork.cs [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+
+        private DbContext _dbContext;
+
+        public UnitOfWork(DbContext dbContext)
+        {
+            _dbContext = dbContext;
+        }
+
+
+        public IEmployeeRepository EmployeeDataRepository
```
Comment (GRustamyan-ST) [inline src/IdentityManager/Travely.IdentityManager.Repository/UnitOfWork.cs:23]: See comment on IUnitOfWork

**Review by GRustamyan-ST — CHANGES_REQUESTED** [review]

File: src/IdentityManager/Travely.IdentityManager.IRepository/Travely.IdentityManager.IRepository.csproj [hunk 2]
```diff
@@ -0,0 +1,15 @@
+<Project Sdk="Microsoft.NET.Sdk">
```
[the remark this answers is not in the document]
Comment (GhTsovinar) [reply src/IdentityManager/Travely.IdentityManager.IRepository/Travely.IdentityManager.IRepository.csproj:1]: done

File: src/IdentityManager/Packages.props [hunk 3]
```diff
@@ -6,6 +6,10 @@
   <ItemGroup Label="Nupkg Versions">
     <PackageReference Update="IdentityServer4" Version="4.1.1" />
     <PackageReference Update="IdentityServer4.AccessTokenValidation" Version="3.0.1" />
+    <PackageReference Update="IdentityServer4.EntityFramework" Version="4.1.1" />
```
[the remark this answers is not in the document]
Comment (GhTsovinar) [reply src/IdentityManager/Packages.props:9]: removed it

File: src/IdentityManager/Travely.IdentityManager.IRepository/IAgencyRepository.cs [hunk 4]
```diff
@@ -0,0 +1,8 @@
+﻿using Travely.IdentityManager.Repository.Model.AppEntities;
+
+namespace Travely.IdentityManager.IRepository
```
[the remark this answers is not in the document]
Comment (GhTsovinar) [reply src/IdentityManager/Travely.IdentityManager.IRepository/IAgencyRepository.cs:3]: done

File: src/IdentityManager/Travely.IdentityManager.Repository/BaseRepository.cs [hunk 5]
```diff
@@ -0,0 +1,54 @@
+﻿using Microsoft.EntityFrameworkCore;
+using System;
+using System.Linq;
+using System.Threading.Tasks;
+using Travely.IdentityManager.IRepository;
+
+namespace Travely.IdentityManager.Repository
+{
+    public class BaseRepository<TEntity, U> : IBaseRepository<TEntity> where TEntity : class, new()
```
[the remark this answers is not in the document]
Comment (GhTsovinar) [reply src/IdentityManager/Travely.IdentityManager.Repository/BaseRepository.cs:9]: done

File: src/IdentityManager/Travely.IdentityManager.Repository/UnitOfWork.cs [hunk 6]
```diff
@@ -0,0 +1,113 @@
...
+{
+    public class UnitOfWork : IUnitOfWork
+    {
+        private IEmployeeRepository _employeeRepository;
+        private IAgencyRepository _agencyRepository;
+        private IUserRepository _userRepository;
+
+        private DbContext _dbContext;
+
+        public UnitOfWork(DbContext dbContext)
```
[the remark this answers is not in the document]
Comment (GhTsovinar) [reply src/IdentityManager/Travely.IdentityManager.Repository/UnitOfWork.cs:17]: done

File: src/IdentityManager/Travely.IdentityManager.Repository.Model/AppEntities/Agency.cs [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+using System.Collections.Generic;
+using System.Text;
+
+namespace Travely.IdentityManager.Repository.Model.AppEntities
+{
+    public class Agency
+    {
+        public Agency()
+        {
+            Employees = new HashSet<Employee>();
```
[the remark this answers is not in the document]
Comment (GhTsovinar) [reply src/IdentityManager/Travely.IdentityManager.Repository.Model/AppEntities/Agency.cs:11]: done

File: src/IdentityManager/Travely.IdentityManager.IRepository/IBaseRepository.cs [hunk 8]
```diff
@@ -0,0 +1,17 @@
...
+using System.Threading.Tasks;
+
+namespace Travely.IdentityManager.IRepository
+{
+    public interface IBaseRepository<TEntity> where TEntity : class, new()
+    {
+        TEntity Add(TEntity entity);
+        TEntity Update(TEntity entity);
+        IQueryable<TEntity> GetAll();
+        Task<TEntity> GetByIdAsync(int id);
```
[the remark this answers is not in the document]
Comment (GhTsovinar) [reply src/IdentityManager/Travely.IdentityManager.IRepository/IBaseRepository.cs:15]: done

File: src/IdentityManager/Travely.IdentityManager.IRepository/IBaseRepository.cs [hunk 9]
```diff
@@ -0,0 +1,17 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Linq.Expressions;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Travely.IdentityManager.IRepository
+{
+    public interface IBaseRepository<TEntity> where TEntity : class, new()
```
[the remark this answers is not in the document]
Comment (GhTsovinar) [reply src/IdentityManager/Travely.IdentityManager.IRepository/IBaseRepository.cs:10]: needs to be discussed

File: src/IdentityManager/Travely.IdentityManager.IRepository/IUnitOfWork.cs [hunk 10]
```diff
@@ -0,0 +1,13 @@
+﻿using System;
+using System.Threading.Tasks;
+
+namespace Travely.IdentityManager.IRepository
+{
+    public interface IUnitOfWork : IDisposable
+    {
+        IUserRepository UserRepository { get; }
```
[the remark this answers is not in the document]
Comment (GhTsovinar) [reply src/IdentityManager/Travely.IdentityManager.IRepository/IUnitOfWork.cs:8]: needs to be discussed

File: src/IdentityManager/Travely.IdentityManager.Repository/UnitOfWork.cs (hunk 1 above)
Comment (GhTsovinar) [reply src/IdentityManager/Travely.IdentityManager.Repository/UnitOfWork.cs:23]: needs to be discussed

File: src/IdentityManager/Travely.IdentityManager.IRepository/IBaseRepository.cs (hunk 9 above)
[the remark this answers is not in the document]
Comment (GhTsovinar) [reply src/IdentityManager/Travely.IdentityManager.IRepository/IBaseRepository.cs:10]: done

File: src/IdentityManager/Travely.IdentityManager.IRepository/IUnitOfWork.cs (hunk 10 above)
[the remark this answers is not in the document]
Comment (GhTsovinar) [reply src/IdentityManager/Travely.IdentityManager.IRepository/IUnitOfWork.cs:8]: done

File: src/IdentityManager/Travely.IdentityManager.Repository/UnitOfWork.cs (hunk 1 above)
Comment (GhTsovinar) [reply src/IdentityManager/Travely.IdentityManager.Repository/UnitOfWork.cs:23]: done

**Comment by GhTsovinar** [comment]
created another branch

**Pull request closed without merging**